### ISSUE TITLE
Command line option to limit the number of renewals

### ISSUE THREAD

**Issue by mattkolb** [certbot/certbot#10083]
I have a large number of certs to the point where "certbot renew" takes days to run which makes scheduling auto renewal problematic. Having a command line option to limit the number of renewals would make it so that renewals could be done in smaller batches. 

For instance, if I have 3000 certs to renew, I could automate the renewal with the following cron:
0 0,12 * * * /opt/certbot/bin/python -c 'import random; import time; time.sleep(random.random() * 3600)' && sudo certbot renew -q --limit 100

Note the command line option "--limit 100". That would do a batch of 100 every day and spread out the renewals over a 30 day period.

It would make load on the renewal servers more predictable. I can also see this feature request having additional use cases.

**Comment by mattkolb** [comment]
https://github.com/certbot/certbot/pull/10163

**Comment by wgreenberg** [comment]
thanks for the PR, but we'd like to address the underlying problem (certbot taking days to run `certbot renew`) rather than introducing a kludge to work around it.

could you explain how `certbot renew` is taking days to run? are all 3000 certs renewing at the same time?

**Comment by mattkolb** [comment]
Correct. It's actually a little over 6000 renewals. I just used 3000 to make the math easy for the example. 

So, right now, I have a little over 6000 certs on a single server. They expire between 2025-04-28 and 2025-04-29 (80 to 81 days). When they get down to the 30 day point, all 6000 need to renew at the same time. 

From my understanding, certbot is not made to do renewals in parallel. So I'm unsure of how to speed up the process. 

Without the ability to speed up the process, my thought was to break the process up into batches so they do not need to all renew at the same time. That was the intent of the PR.

Ideas?

**Comment by mattkolb** [comment]
Any ideas besides the submitted pull request?

**Comment by mattkolb** [comment]
@wgreenberg @ohemorange @bmw 

We have another renewal coming up. Would love some more feedback. I'm more than willing to contribute, but I'm unsure of the solution. I thought the attached pull request would fix this issue as well as have additional uses. For instance, I would have used it when testing new setups. Even for a dozen certs, it would be nice to be able to test 1 and get 1 error instead of testing and having all dozen certs generate errors.

**Comment by zoracon** [comment]
I know this isn't a solution, but it might help to open up a discussion in https://community.letsencrypt.org/. 1. Because someone else may have come across this and possibly found a solution OR at the very least get visibility on this issue to get more ideas on what a solution could look like.

**Comment by mattkolb** [comment]
Thank you for the suggestion. I made a post over there.

**Comment by mattkolb** [comment]
Here is the post: https://community.letsencrypt.org/t/need-suggestions-for-renewing-thousands-of-certificates/235311/5

Suggestions ranged: condemning certbot, write a limiter, and suggestions discouraged by the person suggesting.

Doesn't the included pull request seem like the best solution?

Beyond my use case, I would have found this option quite useful when originally implementing certbot and doing testing. I'm sure others would find a use for it as well...
